public/doc/openapi-3-v1.0.0.json

```diff
@@ -1422,6 +1422,163 @@
           }
         }
       }
+    },
+    "/settings": {
+      "get": {
+        "tags": [
+          "Settings"
+        ],
+        "summary": "List Tenant Settings",
+        "description": "List Tenant Settings",
+        "operationId": "listSettings",
+        "responses": {
+          "200": {
+            "description": "A json object with key/value pairs of tenant settings",
+            "content": {
+              "application/json": {
```

Comment:
@lindgrenj6 Should this be a using a json schema so that the UI can render the settings without any issues and send the API an object which has has key value pairs. The icon would just be a text area. The workflow could be a drop down with the list of approved workflows. We have done something similar to this for the Provider control parameters. We deliver a JSON schema payload to the UI, it fills in all the value and we store it as a JSONB column.

https://github.com/ManageIQ/catalog-api/tree/master/schemas/json 

contains 2 JSON schema examples we use the provider control parameters today but the web hook one is more complete and shows the power of JSON schema. We could have a Tenant Settings Schema in this folder and allow the UI to fetch this payload like here
https://github.com/ManageIQ/catalog-api/blob/5927669b8e1ba943f7bc69c5244da402615d9170/public/doc/openapi-3-v1.0.0.json#L825
The response object could contain 2 pieces of information the Current value object and the JSON schema. In the above example we are only delivering the JSON schema because it is directly merged into the Order Item Provider control parameters and never needs to be edited (not yet at least)

Reply:
@mkanoor I added an implementation of this, let me know what you think. 
Currently it's returning like so now: 
```
{
	"current": { 
 		"setting": "value"
	},
	"schema": { 
		"schema_setting": 2
	}
}
```
so the UI will have the user settings as well as the schema to build their form from. I have to add the dynamic replacement of values in the schema still, but I think I'll get that done today.

Reply:
@lindgrenj6 Looks good. @Hyperkid123 is adding some dynamic content logic into the JSON schema once he implements it then we won't need to update the schema with the new values.

This is what the proposal looked like for dynamic content, not finalized yet, I have added your approval API call

```
content                   dynamic
apiClass                  approval
operationId             listWorkflows
parameters              []
resultAttributes        ["name", "id"]
resultSortBy            ["name"]
resultSortOrder       ascending
select                       id
```

Reply:
The component is ready no we just need to map this (if ready) schema to it.

Reply:
@Hyperkid123 cool, this one is ready as of right now I think. is there documentation to set up the mappings?

Reply:
@lindgrenj6 i will have to map it in the client in the component. By mapping i mean picking correct api client, collection and method to fetch the data.

Reply:
@Hyperkid123 :+1: madhu seems to have provided that information above, so if you need anything else let us know!

Reply:
Yep i know :wink:. Once its merged i will start playing around with it